src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -339,20 +417,32 @@ public void MonitoringSurvivedProcessMemorySize()
         [Fact]
         public void MonitoringTotalAllocatedMemorySize()
         {
-            Assert.Throws<InvalidOperationException>(() => { var t = AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize; } );
+            RemoteInvoke(() => {
+                Assert.Throws<InvalidOperationException>(() => {
+                    var t = AppDomain.CurrentDomain.MonitoringTotalAllocatedMemorySize;
+                });
+                return SuccessExitCode;
+            }).Dispose();
         }
 
         [Fact]
         public void MonitoringTotalProcessorTime()
         {
-            Assert.Throws<InvalidOperationException>(() => { var t = AppDomain.CurrentDomain.MonitoringTotalProcessorTime; } );
+            Assert.Throws<InvalidOperationException>(() => {
+                var t = AppDomain.CurrentDomain.MonitoringTotalProcessorTime;
+            } );
         }
 
 #pragma warning disable 618
         [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
         public void GetCurrentThreadId()
         {
-            Assert.True(AppDomain.GetCurrentThreadId() == Environment.CurrentManagedThreadId);
+            RemoteInvoke(() =>
```

Comment:
I'm not sure. There is almost no cost of running this remotely so not sure if this is worth investigating